src/pages/vehicleList.tsx

```diff
@@ -0,0 +1,77 @@
+import { useContext } from "react"
+import { CarsContext } from "../components/providers/carsProvider"
+import { MotoContext } from "../components/providers/motoProvider";
+import { DataGrid, GridActionsCellItem, GridColDef, GridRowId } from '@mui/x-data-grid';
+import { Box, Typography } from "@mui/material";
+import { Car, engineTypes, Motorcycle } from "../lib/types";
+import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
+import FavoriteIcon from '@mui/icons-material/Favorite';
+import DeleteIcon from '@mui/icons-material/Delete';
+import TwoWheeler from "@mui/icons-material/TwoWheeler";
+import DirectionsCarIcon from '@mui/icons-material/DirectionsCar';
+import { LocaleContext } from "../components/providers/localeProvider";
+
+export function VehicleList() {
+    const { cars, setCars, deleteCar } = useContext(CarsContext)
+    const { Motorcycles, setMotorcycles, deleteMotorcycle } = useContext(MotoContext)
+    const { translation } = useContext(LocaleContext)
+    const f = translation.form
+    function handleRowUpdate(newRow: any) {
+        if (newRow.id === undefined) return
+        const updatedRow = { ...newRow, isNew: false };
+        if (newRow.id as number < cars.length)
+            setCars(cars.map((car, i) => i === newRow.id ? newRow as Car : car))
+        else
+            setMotorcycles(Motorcycles.map((moto, i) => i === newRow.id - cars.length ? newRow as Motorcycle : moto))
+        return updatedRow
+    }
+
+    function handleDelete(i: GridRowId){
+        i = i as number
+        if (i < cars.length)
+            deleteCar(i)
+        else
+            deleteMotorcycle(i - cars.length)
+    }
```

Comment:
Maybe if you pass a type (car/moto) in addition to id, it will be more convenient to delete ?